### PR TITLE
project is ready for a code review

### PR DESCRIPTION
**Pull request by misterSokor**

### CODE REVIEW

File: src/main/java/com/store/litflix/controller/BookController.java [hunk 1]
```diff
@@ -26,12 +30,22 @@ BookDto createBook(@RequestBody CreateBookRequestDto requestDto) {
     }
 
     @GetMapping("/{id}")
-    Book getBook(@PathVariable Long id) {
+    Optional<Book> getBook(@PathVariable Long id) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/store/litflix/controller/BookController.java:33]:
```suggestion
    BookDto getBook(@PathVariable Long id) {
```

File: src/main/java/com/store/litflix/controller/BookController.java [hunk 2]
```diff
@@ -26,12 +30,22 @@ BookDto createBook(@RequestBody CreateBookRequestDto requestDto) {
...
         return bookService.findById(id);
     }
 
     @GetMapping
     List<Book> getAllBooks() {
```
Comment (Elena-Bruyako) [inline src/main/java/com/store/litflix/controller/BookController.java:38]:
```suggestion
    List<BookDto> getAllBooks() {
```

File: src/main/java/com/store/litflix/repository/BookRepository.java [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
-public interface BookRepository {
-    Book save(Book book);
-
-    List<Book> findAll();
-
-    Book findById(Long id);
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface BookRepository extends JpaRepository<Book, Long> {
+//    Book save(Book book);
```
Comment (Elena-Bruyako) [inline src/main/java/com/store/litflix/repository/BookRepository.java:7]: remove comments

File: src/main/java/com/store/litflix/service/impl/BookServiceImpl.java [hunk 4]
```diff
@@ -30,7 +33,25 @@ public List<Book> findAll() {
     }
 
     @Override
-    public Book findById(Long id) {
+    public Optional<Book> findById(Long id) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/store/litflix/service/impl/BookServiceImpl.java:36]:
```suggestion
    public BookDto findById(Long id) {
```

File: src/main/java/com/store/litflix/service/impl/BookServiceImpl.java [hunk 5]
```diff
@@ -30,7 +33,25 @@ public List<Book> findAll() {
...
+    public BookDto updateBook(Long id, UpdateBookRequestDto requestDto) {
+        Optional<Book> optionalBook = bookRepository.findById(id);
+        if (optionalBook.isPresent()) {
+            Book book = optionalBook.get();
+            Book modelFromDto = bookMapper.updateModelFromDto(requestDto, book);
+            Book updatedBook = bookRepository.save(modelFromDto);
+            return bookMapper.toDto(updatedBook);
+        } else {
+            throw new EntityNotFoundException("Book not found with id " + id);
+        }
```
Comment (Elena-Bruyako) [inline src/main/java/com/store/litflix/service/impl/BookServiceImpl.java:50]:
```suggestion
        Book book = bookRepository.findById(id)
                .orElseThrow(() -> new EntityNotFoundException("Book not found with id " + id);
        bookMapper.updateBookFromDto(bookDto, book);
        bookRepository.save(book);
        return bookMapper.toDto(book);
```

File: src/main/resources/db/changelog/changes/02-add-isDeleted-to-books-table.yaml [hunk 6]
```diff
@@ -0,0 +1,14 @@
+databaseChangeLog:
+  - changeSet:
+      id: add-isDeleted-to-books-table.yaml
+      author: Peter Sokor
+      changes:
+        - addColumn:
+            tableName: books
+            columns:
+              - column:
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/changes/02-add-isDeleted-to-books-table.yaml:9]: replace to existing changelog 01

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comment

File: src/main/resources/db/changelog/changes/02-add-isDeleted-to-books-table.yaml (hunk 6 above)
Comment (misterSokor) [reply src/main/resources/db/changelog/changes/02-add-isDeleted-to-books-table.yaml:9]: Could you please clarify? I have to delete  01-create-books-table.yaml file and to replace it with 02-add-isDeleted-to-books-table.yaml?

File: src/main/java/com/store/litflix/controller/BookController.java (hunk 1 above)
Comment (Elena-Bruyako) [reply src/main/java/com/store/litflix/controller/BookController.java:33]: not fixed

File: src/main/java/com/store/litflix/controller/BookController.java [hunk 7]
```diff
@@ -26,12 +30,22 @@ BookDto createBook(@RequestBody CreateBookRequestDto requestDto) {
...
         return bookService.findAll();
     }
+
+    @PutMapping("/{id}")
+    BookDto updateBook(@PathVariable Long id, @RequestBody UpdateBookRequestDto requestDto) {
+        return bookService.updateBook(id, requestDto);
+    }
+
+    @DeleteMapping("/{id}")
+    void deleteById(@PathVariable Long id) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/store/litflix/controller/BookController.java:48]: make all endpoints public

File: src/main/java/com/store/litflix/service/impl/BookServiceImpl.java [hunk 8]
```diff
@@ -20,17 +23,39 @@ public class BookServiceImpl implements BookService {
     public BookDto save(CreateBookRequestDto requestDto) {
         Book book = bookMapper.toModel(requestDto);
         Book savedBook = bookRepository.save(book);
```
Comment (Elena-Bruyako) [inline src/main/java/com/store/litflix/service/impl/BookServiceImpl.java:25]:
```suggestion
        bookRepository.save(book);
```

File: src/main/java/com/store/litflix/service/impl/BookServiceImpl.java [hunk 9]
```diff
@@ -20,17 +23,39 @@ public class BookServiceImpl implements BookService {
...
-        BookDto bookDto = bookMapper.toDto(savedBook);
-        return bookDto;
+        return bookMapper.toDto(savedBook);
```
Comment (Elena-Bruyako) [inline src/main/java/com/store/litflix/service/impl/BookServiceImpl.java:26]:
```suggestion
        return bookMapper.toDto(book);
```

File: src/main/java/com/store/litflix/service/impl/BookServiceImpl.java (hunk 4 above)
Comment (Elena-Bruyako) [reply src/main/java/com/store/litflix/service/impl/BookServiceImpl.java:38]: not fixed

File: src/main/java/com/store/litflix/service/impl/BookServiceImpl.java [hunk 10]
```diff
@@ -20,17 +23,39 @@ public class BookServiceImpl implements BookService {
...
+        if (optionalBook.isPresent()) {
+            Book book = bookRepository.findById(id)
+                    .orElseThrow(() -> new EntityNotFoundException("Book not "
+                                                                   + "found with id " + id));
+            bookMapper.updateModelFromDto(requestDto, book);
+            bookRepository.save(book);
+            return bookMapper.toDto(book);
+        } else {
+            throw new EntityNotFoundException("Book not found with id " + id);
+        }
```
Comment (Elena-Bruyako) [inline src/main/java/com/store/litflix/service/impl/BookServiceImpl.java:54]:
```suggestion
        Book book = bookRepository.findById(id)
                    .orElseThrow(() -> new EntityNotFoundException("Book not "
                                                                   + "found with id " + id));
        bookMapper.updateModelFromDto(requestDto, book);
        bookRepository.save(book);
        return bookMapper.toDto(book);
```

File: src/main/resources/db/changelog/changes/02-add-isDeleted-to-books-table.yaml (hunk 6 above)
Comment (Elena-Bruyako) [reply src/main/resources/db/changelog/changes/02-add-isDeleted-to-books-table.yaml:9]: Add column is_deleted to existing 01 changelog
If you have questions please ask in chat

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/com/store/litflix/controller/BookController.java (hunk 2 above)
Comment (misterSokor) [reply src/main/java/com/store/litflix/controller/BookController.java]: the same thing here

File: src/main/resources/db/changelog/changes/02-add-isDeleted-to-books-table.yaml (hunk 6 above)
Comment (misterSokor) [reply src/main/resources/db/changelog/changes/02-add-isDeleted-to-books-table.yaml:9]: Thanks

File: src/main/java/com/store/litflix/service/impl/BookServiceImpl.java (hunk 4 above)
Comment (Elena-Bruyako) [reply src/main/java/com/store/litflix/service/impl/BookServiceImpl.java:38]: findById method should return BookDto on Service layer
i've never suggested to return Optional<Book> from this method in any commits.

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comment

**Review by Elena-Bruyako — APPROVED** [review]

**Pull request merged**